ShelterModule/Models/Announcements/AnnouncementResponse.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace ShelterModule.Models.Announcements;
+
+public sealed class AnnouncementResponse
+{
+    public required Guid Id { get; init; }
+
+    public required Guid AuthorId { get; init; }
+
+    public required Guid PetId { get; init; }
+
+    public required string Title { get; init; } = null!;
+
+    public required string Description { get; init; } = null!;
+
+    public required DateTime CreationDate { get; init; }
+
+    public DateTime? ClosingDate { get; init; }
```

Comment:
Also, we can have `required` keyword here even if this property is nullable. It ensures that we need to assign something to this property when we create this object, even if it's `null`. But that's not too important.